### ISSUE TITLE
Create a Drupal theme

### ISSUE THREAD

**Issue by fureigh** [18F/web-design-standards#852]
**User story:**
As an administrator of a [Drupal](https://www.drupal.org) site, government or otherwise, I want to be able to efficiently assess or implement the [U.S. Web Design Standards](https://playbook.cio.gov/designstandards/) (USWDS) so that deciding whether to adopt and how to use them will entail minimal hassle and no custom development.

**Background:**
As of 2012, an estimated 24 percent of U.S. government websites (including whitehouse.gov, most famously) were Drupal sites. That number has only continued to increase.

Just as the [Bootstrap](http://drupal.org/project/bootstrap) theme bridges the gap between Drupal and the [Bootstrap Framework](http://getbootstrap.com), a USWDS theme will make it simpler and easier for Drupal site administrators to use the U.S. Web Design Standards.

**Technical notes:**
 * In practice, this will entail the creation of two themes, one for D8 and one for D7.
 * I plan to start this project's development on GitHub, eventually mirroring it on drupal.org.

**Outstanding questions:**
 * Who else wants to help/test/etc.?

**Comment by fureigh** [comment]
Initial repo: https://github.com/18F/web-design-standards-drupal-d7

**Comment by fureigh** [comment]
Great minds think alike: just reached out to AlexB-appnovation about [his sandbox](https://www.drupal.org/sandbox/alexb-appnovation/2597672) and how best to collaborate.

**Comment by jonraedeke** [comment]
I'd love to help!

**Comment by fureigh** [comment]
@jonraedeke Fantastic! Good to talk with you earlier. :)

Sounds like Alex has also started working on a Drupal 7 theme, so I'll continue toward a Drupal 8 version and we'll be comparing notes.

**Comment by fureigh** [comment]
Looping in @AlexBrown-appnovation. :wave:

**Comment by fureigh** [comment]
Let's leave this issue open until a theme has been released, but in the spirit of leaving clear breadcrumbs: all further conversation and development is happening in https://github.com/18F/web-design-standards-drupal.

**Comment by pjurquilla** [comment]
This would be awesome.

**Comment by pjurquilla** [comment]
See also: Federal web design standards to become ready-to-use Drupal, WordPress themes 
http://www.fiercegovernmentit.com/story/federal-web-design-standards-become-ready-use-drupal-wordpress-themes/2015-12-03?utm_medium=nl&utm_source=internal

**Comment by bbertucc** [comment]
@fureigh- is there a repo for the WordPress theme? I'm ready to start building if none exist.

**Comment by fureigh** [comment]
@pjurquilla Neat, thanks for sharing!

@bbertucc Not that I'm aware of. I'd say go for it and work in the open so other people can find you. (ccing @gboone, who also mentioned WordPress interest)

**Comment by fureigh** [comment]
@bbertucc To rephrase that a little more clearly: it's awesome that you're ready to start building! Please do, and share your work. :)

**Comment by bbertucc** [comment]
thanks @fureigh  .. I'm going to play with some ideas locally. I'll definitely share and keep it open when I'm ready to create the repo.

**Comment by iris-i** [comment]
It was great speaking with you @fureigh! @AlexBrown-appnovation Civicactions would love to be part of the drupal7 work on this. We have a week assigned to this project starting on 12/14 - 12/18. That amounts to about 80 hours of work between I & @kevwalsh and 40 hrs PM time allocated to @stevecurtis. We're thinking of getting on a call with you this week, to figure out how best to start working on this so we can hit the ground running next week. Please let us know if you will be free to connect and chat. Looking forward!

**Comment by AlexBrown-appnovation** [comment]
Hey @iris-i ! That's great, glad to have you on board!

Yes feel free to email me and we can chat further - alex@appnovation.com.

**Comment by fureigh** [comment]
Hooray! Plenty of work to go around. Feel free to loop me into that call; happy to see how I can best support.

**Comment by fureigh** [comment]
Looping in @jenter, a front-end architect who's been doing some related work for a client. Want to make [placeholder issues](http://www.github.com/18F/web-design-standards-drupal/issues) or work-in-progress pull requests for anything you're particularly excited to contribute back?

**Comment by gboone** [comment]
Here's a start on a WordPress theme. Just a start :wink: https://github.com/gboone/uswds-theme

**Comment by jenter** [comment]
@fureigh great chatting with you yesterday. 

Here's some initial notes but we can correspond about the details:
* I have a couple of panel layouts already I can cleanup and commit in a PR 
* I have some preprocess functions, but wanted to get your thoughts first. Are you planning having these used when installed OR are you going to have a collection of commented-out examples?
* We are using a simple gulp file http://screencast.com/t/jC8L838N but wanted to get your thoughts on any advanced task running (linting, img compression,etc)

**Comment by bbertucc** [comment]
@gboone - I too started a repo for a wp theme .. https://github.com/bbertucc/us-web-design-standards-wp ... currently working on a _s and sage integration aswell .. will be syncing them sometime this week .. excited to watch your progress, and maybe work together ..

**Comment by gboone** [comment]
@bbertucc Coo! I'll check it out. It's been a while since I made a theme so I may just fork yours and work together rather than in parallel.

**Comment by iris-i** [comment]
Hey @jenter! I started an issue to document gulp plugins we could add to a task, would you like to put your thoughts in here? https://github.com/18F/web-design-standards-drupal/issues/25
cc: @fureigh & @AlexBrown-appnovation

**Comment by jflowers45** [comment]
Hi all,
  @bbgvisualjournalist, @gigifrias and I are also working on a standards-inspired theme that will tie together a few federal WordPress sites and we used _s as our starting point.  Our repo is at https://github.com/BBGInnovate/bbgWPtheme - mainly just chiming in to say hello but will share more as we have it.

**Comment by bbertucc** [comment]
@bbgvisualjournalist @gigifrias @gboone @fureigh  - I just launched a separate issue thread to address the need of a USWDS WordPress theme: https://github.com/18F/web-design-standards/issues/926

**Comment by fureigh** [comment]
@bbertucc :+1: Thanks for doing that!

**Comment by fureigh** [comment]
@jenter: Thanks for those thoughts!

Responses in order:

 * **Panel layouts:** Definitely interested in seeing the panel layouts. Let's bookmark the question of whether to let the theme require Panels, but at minimum these could be a valuable enhancement.
 * **Preprocess functions:** These would be great to have first! The preprocess functions sound most foundationally relevant because they'll apply the grid classes, etc. throughout. The guiding principle here has been to err on the side of out-of-the-box functionality, so while commented-out examples would be good as part of code documentation, let's go ahead and use those preprocess functions by default. We can later expand on that to allow users to enable and disable that extra markup through the UI.
 * **Gulp setup:** :+1: I just merged the one @AlexBrown-appnovation submitted in https://github.com/18F/web-design-standards-drupal/pull/20, but @iris-i is continuing the conversation about linters &c. in https://github.com/18F/web-design-standards-drupal/issues/25.

**Comment by juliaelman** [comment]
Thank you @fureigh for creating the Drupal theme! Closing this issue and encourage folks to follow along here: https://github.com/18F/web-design-standards-drupal

**Comment by subtera745** [comment]
@fureigh is the Drupal theme still being updated and maintained by 18f? It unfortunately is in a beta stage for Drupal 8, I was curious if you by chance knew when it would be made a production release at its current stage?
